bottlerocket/agents/src/workload.rs

```diff
@@ -0,0 +1,182 @@
+use crate::error;
+use crate::sonobuoy::process_sonobuoy_test_results;
+use bottlerocket_types::agent_config::{WorkloadConfig, SONOBUOY_RESULTS_FILENAME};
+use log::{info, trace};
+use model::TestResults;
+use snafu::{ensure, ResultExt};
+use std::fs::File;
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+const SONOBUOY_BIN_PATH: &str = "/usr/bin/sonobuoy";
+
+/// Runs the workload conformance tests according to the provided configuration and returns a test
+/// result at the end.
+pub async fn run_workload(
+    kubeconfig_path: &str,
+    workload_config: &WorkloadConfig,
+    results_dir: &Path,
+) -> Result<TestResults, error::Error> {
+    info!("Processing workload test plugins");
+    let mut plugin_test_args: Vec<String> = Vec::new();
+    for (id, plugin) in workload_config.plugins.iter().enumerate() {
+        info!("Initializing test {}-{}", id, plugin.name);
+        let output = Command::new(SONOBUOY_BIN_PATH)
+            .arg("gen")
+            .arg("plugin")
+            .arg("--name")
+            .arg(plugin.name.clone())
+            .arg("--image")
+            .arg(plugin.image.clone())
+            .output()
+            .context(error::WorkloadProcessSnafu)?;
+        ensure!(
+            output.status.success(),
+            error::WorkloadPluginSnafu {
+                plugin: plugin.name.clone()
+            }
+        );
+
+        // Write out the output to a file we can reference later
+        let file_name = format!("{}-plugin.yaml", plugin.name);
+        let plugin_yaml = PathBuf::from(".")
+            .join(&file_name)
+            .canonicalize()
+            .context(error::BadPathSnafu { path: &file_name })?;
+        let mut f = File::create(&plugin_yaml).context(error::FileWriteSnafu {
+            path: plugin_yaml.display().to_string(),
+        })?;
+        f.write_all(&output.stdout)
+            .context(error::WorkloadProcessSnafu)?;
+
+        // Add plugin to the arguments to be passed to sonobuoy run
+        plugin_test_args.append(&mut vec![
+            "--plugin".to_string(),
+            plugin_yaml.display().to_string(),
+        ]);
+    }
+
+    info!("Running workload");
+    let kubeconfig_arg = vec!["--kubeconfig", kubeconfig_path];
+    let output = Command::new(SONOBUOY_BIN_PATH)
+        .args(kubeconfig_arg.to_owned())
+        .arg("run")
+        .arg("--wait")
+        .arg("--namespace")
+        .arg("testsys-workload")
```

Comment:
Does this mean we could run a normal `sonobuoy` test and `workload` test on the same cluster at the same time? (Since they would be in different namespaces)

Reply:
Ran additional tests and verified, see comment above in `sonobuoy.rs`.